fix(components): add class name and disable style when disabling upload

ma-shuo opened the pull request:
closed #17746

Please make sure these boxes are checked before submitting your PR, thank you!

- [x] Make sure you follow contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [x] Make sure you are merging your commits to `dev` branch.
- [x] Add some descriptions and refer to relative issues for your PR.

ma-shuo committed:
fix(components): add class name and disable style when disabling upload

closed #17746

ma-shuo committed:
Merge branch 'dev' into dev

ma-shuo committed:
style(components): hover and focus styles when the upload is disabled

ma-shuo committed:
style(components): hover and focus styles when the upload is disabled

closed #17746

ma-shuo committed:
Merge remote-tracking branch 'origin/dev' into dev

ma-shuo committed:
Merge branch 'dev' into dev

ma-shuo committed:
Merge branch 'dev' into dev

btea commented on packages/components/upload/src/upload-content.vue:
```diff
@@ -1,6 +1,11 @@
+      ns.is('drag', drag),
+      ns.is('disabled', disabled),
+    ]"
     tabindex="0"
```
Maybe it would be better to set `tabindex` to `-1` when the setting is disabled, so that you don't need to add focus related styles.

ma-shuo replied:
After setting the tabindex to -1, I found that the mouse could also focus, and then I solved the problem by setting "pointer events" to "none" and deleting the excess CSS

ma-shuo committed:
style(components): simplifiser CSS-stiler

When disabling upload, use "pointer events" to unfocus the mouse and set "tabindex" to -1 to unfocus
the tab

closed #17746

ma-shuo committed:
Merge remote-tracking branch 'origin/dev' into dev

ma-shuo committed:
Merge branch 'dev' into dev

btea reviewed (approved):
LGTM

ma-shuo committed:
Merge branch 'dev' into dev

ma-shuo committed:
Merge branch 'dev' into dev

ma-shuo committed:
Merge branch 'dev' into dev

ma-shuo commented:
> LGTM


Hello, could you please run the workflow
<img width="1251" alt="image" src="https://github.com/user-attachments/assets/5d134055-61b8-4570-a42c-049a806cff9d">

ma-shuo commented:
> LGTM

Hello, this is my first submission. How can I merge it into the repository after review

btea commented:
Don't worry, it usually takes two people to review it before it can be merged. Please be patient and wait for others to review it.

tolking commented:
When disabling the upload component, the mouse cursor should display the disabled style (cursor: not-allowed), but it doesn't seem to be effective now [Link](https://element-plus.run/#eyJBcHAudnVlIjoiPHRlbXBsYXRlPlxuICA8ZWwtaW5wdXQgZGlzYWJsZWQgLz5cbiAgPGVsLXVwbG9hZFxuICAgIGNsYXNzPVwidXBsb2FkLWRlbW9cIlxuICAgIGRyYWdcbiAgICBhY3Rpb249XCJodHRwczovL3J1bi5tb2NreS5pby92My85ZDA1OWJmOS00NjYwLTQ1ZjItOTI1ZC1jZTgwYWQ2YzRkMTVcIlxuICAgIG11bHRpcGxlXG4gICAgOmRpc2FibGVkPVwidHJ1ZVwiXG4gID5cbiAgICA8ZWwtaWNvbiBjbGFzcz1cImVsLWljb24tLXVwbG9hZFwiPjx1cGxvYWQtZmlsbGVkIC8+PC9lbC1pY29uPlxuICAgIDxkaXYgY2xhc3M9XCJlbC11cGxvYWRfX3RleHRcIj5cbiAgICAgIERyb3AgZmlsZSBoZXJlIG9yIDxlbT5jbGljayB0byB1cGxvYWQ8L2VtPlxuICAgIDwvZGl2PlxuICAgIDx0ZW1wbGF0ZSAjdGlwPlxuICAgICAgPGRpdiBjbGFzcz1cImVsLXVwbG9hZF9fdGlwXCI+XG4gICAgICAgIGpwZy9wbmcgZmlsZXMgd2l0aCBhIHNpemUgbGVzcyB0aGFuIDUwMGtiXG4gICAgICA8L2Rpdj5cbiAgICA8L3RlbXBsYXRlPlxuICA8L2VsLXVwbG9hZD5cbiAgPGVsLXVwbG9hZFxuICAgIGNsYXNzPVwiYXZhdGFyLXVwbG9hZGVyXCJcbiAgICBkaXNhYmxlZFxuICAgIGFjdGlvbj1cImh0dHBzOi8vcnVuLm1vY2t5LmlvL3YzLzlkMDU5YmY5LTQ2NjAtNDVmMi05MjVkLWNlODBhZDZjNGQxNVwiXG4gICAgOnNob3ctZmlsZS1saXN0PVwiZmFsc2VcIlxuICA+XG4gICAgPGltZyB2LWlmPVwiaW1hZ2VVcmxcIiA6c3JjPVwiaW1hZ2VVcmxcIiBjbGFzcz1cImF2YXRhclwiIC8+XG4gICAgPGVsLWljb24gdi1lbHNlIGNsYXNzPVwiYXZhdGFyLXVwbG9hZGVyLWljb25cIj48UGx1cyAvPjwvZWwtaWNvbj5cbiAgPC9lbC11cGxvYWQ+XG48L3RlbXBsYXRlPlxuXG48c2NyaXB0IHNldHVwIGxhbmc9XCJ0c1wiPlxuaW1wb3J0IHsgcmVmIH0gZnJvbSAndnVlJztcbmltcG9ydCB7IFVwbG9hZEZpbGxlZCwgUGx1cyB9IGZyb20gJ0BlbGVtZW50LXBsdXMvaWNvbnMtdnVlJ1xuXG5jb25zdCBpbWFnZVVybCA9IHJlZignJylcbjwvc2NyaXB0PlxuXG48c3R5bGUgc2NvcGVkPlxuLmF2YXRhci11cGxvYWRlciAuYXZhdGFyIHtcbiAgd2lkdGg6IDE3OHB4O1xuICBoZWlnaHQ6IDE3OHB4O1xuICBkaXNwbGF5OiBibG9jaztcbn1cbjwvc3R5bGU+XG5cbjxzdHlsZT5cbi5hdmF0YXItdXBsb2FkZXIgLmVsLXVwbG9hZCB7XG4gIGJvcmRlcjogMXB4IGRhc2hlZCB2YXIoLS1lbC1ib3JkZXItY29sb3IpO1xuICBib3JkZXItcmFkaXVzOiA2cHg7XG4gIGN1cnNvcjogcG9pbnRlcjtcbiAgcG9zaXRpb246IHJlbGF0aXZlO1xuICBvdmVyZmxvdzogaGlkZGVuO1xuICB0cmFuc2l0aW9uOiB2YXIoLS1lbC10cmFuc2l0aW9uLWR1cmF0aW9uLWZhc3QpO1xufVxuXG4uYXZhdGFyLXVwbG9hZGVyIC5lbC11cGxvYWQ6aG92ZXIge1xuICBib3JkZXItY29sb3I6IHZhcigtLWVsLWNvbG9yLXByaW1hcnkpO1xufVxuXG4uZWwtaWNvbi5hdmF0YXItdXBsb2FkZXItaWNvbiB7XG4gIGZvbnQtc2l6ZTogMjhweDtcbiAgY29sb3I6ICM4YzkzOWQ7XG4gIHdpZHRoOiAxNzhweDtcbiAgaGVpZ2h0OiAxNzhweDtcbiAgdGV4dC1hbGlnbjogY2VudGVyO1xufVxuPC9zdHlsZT5cbiIsImVsZW1lbnQtcGx1cy5qcyI6ImltcG9ydCB7IGdldEN1cnJlbnRJbnN0YW5jZSB9IGZyb20gJ3Z1ZSdcbmltcG9ydCBFbGVtZW50UGx1cyBmcm9tICdlbGVtZW50LXBsdXMnXG5cbmxldCBpbnN0YWxsZWQgPSBmYWxzZVxuYXdhaXQgbG9hZFN0eWxlKClcblxuZXhwb3J0IGZ1bmN0aW9uIHNldHVwRWxlbWVudFBsdXMoKSB7XG4gIGlmIChpbnN0YWxsZWQpIHJldHVyblxuICBjb25zdCBpbnN0YW5jZSA9IGdldEN1cnJlbnRJbnN0YW5jZSgpXG4gIGluc3RhbmNlLmFwcENvbnRleHQuYXBwLnVzZShFbGVtZW50UGx1cylcbiAgaW5zdGFsbGVkID0gdHJ1ZVxufVxuXG5leHBvcnQgZnVuY3Rpb24gbG9hZFN0eWxlKCkge1xuICBjb25zdCBzdHlsZXMgPSBbJ2h0dHBzOi8vcHJldmlldy0xNzc1My1lbGVtZW50LXBsdXMuc3VyZ2Uuc2gvYnVuZGxlL2luZGV4LmNzcycsICdodHRwczovL3ByZXZpZXctMTc3NTMtZWxlbWVudC1wbHVzLnN1cmdlLnNoL2J1bmRsZS9pbmRleC5jc3MnXS5tYXAoKHN0eWxlKSA9PiB7XG4gICAgcmV0dXJuIG5ldyBQcm9taXNlKChyZXNvbHZlLCByZWplY3QpID0+IHtcbiAgICAgIGNvbnN0IGxpbmsgPSBkb2N1bWVudC5jcmVhdGVFbGVtZW50KCdsaW5rJylcbiAgICAgIGxpbmsucmVsID0gJ3N0eWxlc2hlZXQnXG4gICAgICBsaW5rLmhyZWYgPSBzdHlsZVxuICAgICAgbGluay5hZGRFdmVudExpc3RlbmVyKCdsb2FkJywgcmVzb2x2ZSlcbiAgICAgIGxpbmsuYWRkRXZlbnRMaXN0ZW5lcignZXJyb3InLCByZWplY3QpXG4gICAgICBkb2N1bWVudC5ib2R5LmFwcGVuZChsaW5rKVxuICAgIH0pXG4gIH0pXG4gIHJldHVybiBQcm9taXNlLmFsbFNldHRsZWQoc3R5bGVzKVxufVxuIiwidHNjb25maWcuanNvbiI6IntcbiAgXCJjb21waWxlck9wdGlvbnNcIjoge1xuICAgIFwidGFyZ2V0XCI6IFwiRVNOZXh0XCIsXG4gICAgXCJqc3hcIjogXCJwcmVzZXJ2ZVwiLFxuICAgIFwibW9kdWxlXCI6IFwiRVNOZXh0XCIsXG4gICAgXCJtb2R1bGVSZXNvbHV0aW9uXCI6IFwiQnVuZGxlclwiLFxuICAgIFwidHlwZXNcIjogW1wiZWxlbWVudC1wbHVzL2dsb2JhbC5kLnRzXCJdLFxuICAgIFwiYWxsb3dJbXBvcnRpbmdUc0V4dGVuc2lvbnNcIjogdHJ1ZSxcbiAgICBcImFsbG93SnNcIjogdHJ1ZSxcbiAgICBcImNoZWNrSnNcIjogdHJ1ZVxuICB9LFxuICBcInZ1ZUNvbXBpbGVyT3B0aW9uc1wiOiB7XG4gICAgXCJ0YXJnZXRcIjogMy4zXG4gIH1cbn1cbiIsIlBsYXlncm91bmRNYWluLnZ1ZSI6IjxzY3JpcHQgc2V0dXA+XG5pbXBvcnQgQXBwIGZyb20gJy4vQXBwLnZ1ZSdcbmltcG9ydCB7IHNldHVwRWxlbWVudFBsdXMgfSBmcm9tICcuL2VsZW1lbnQtcGx1cy5qcydcbnNldHVwRWxlbWVudFBsdXMoKVxuPC9zY3JpcHQ+XG5cbjx0ZW1wbGF0ZT5cbiAgPEFwcCAvPlxuPC90ZW1wbGF0ZT5cbiIsImltcG9ydC1tYXAuanNvbiI6IntcbiAgXCJpbXBvcnRzXCI6IHtcbiAgICBcInZ1ZVwiOiBcImh0dHBzOi8vZmFzdGx5LmpzZGVsaXZyLm5ldC9ucG0vQHZ1ZS9ydW50aW1lLWRvbUBsYXRlc3QvZGlzdC9ydW50aW1lLWRvbS5lc20tYnJvd3Nlci5qc1wiLFxuICAgIFwiQHZ1ZS9zaGFyZWRcIjogXCJodHRwczovL2Zhc3RseS5qc2RlbGl2ci5uZXQvbnBtL0B2dWUvc2hhcmVkQGxhdGVzdC9kaXN0L3NoYXJlZC5lc20tYnVuZGxlci5qc1wiLFxuICAgIFwiZWxlbWVudC1wbHVzXCI6IFwiaHR0cHM6Ly9wcmV2aWV3LTE3NzUzLWVsZW1lbnQtcGx1cy5zdXJnZS5zaC9idW5kbGUvaW5kZXguZnVsbC5taW4ubWpzXCIsXG4gICAgXCJlbGVtZW50LXBsdXMvXCI6IFwidW5zdXBwb3J0ZWRcIixcbiAgICBcIkBlbGVtZW50LXBsdXMvaWNvbnMtdnVlXCI6IFwiaHR0cHM6Ly9mYXN0bHkuanNkZWxpdnIubmV0L25wbS9AZWxlbWVudC1wbHVzL2ljb25zLXZ1ZUAyL2Rpc3QvaW5kZXgubWluLmpzXCJcbiAgfSxcbiAgXCJzY29wZXNcIjoge31cbn0iLCJfbyI6eyJzaG93SGlkZGVuIjp0cnVlLCJzdHlsZVNvdXJjZSI6Imh0dHBzOi8vcHJldmlldy0xNzc1My1lbGVtZW50LXBsdXMuc3VyZ2Uuc2gvYnVuZGxlL2luZGV4LmNzcyJ9fQ==)

ma-shuo committed:
style(components): disable upload status and set focus style

closed #17746

ma-shuo committed:
Merge branch 'dev' into dev

ma-shuo commented:
After testing, I found that the focus style must be set. If "pointer events: none;" is used, the mouse style will be the default style instead of disabling the style.

ma-shuo committed:
style(components): disable upload status and set focus style

closed #17746

ma-shuo committed:
Merge branch 'dev' of github.com:ma-shuo/element-plus into dev

ma-shuo committed:
Merge branch 'dev' into dev

tolking commented:
IMO, It is better to keep the disabled Upload component consistent with the disabled Input as much as possible. How about remove ~pointer-events: none~. 

I found that the `disabled` attribute is missing in `<input type="file">` (upload-content.vue). Maybe adding the `:disabled="disabled"` attribute can avoid using `pointer-events: none`.

ma-shuo commented:
> IMO, It is better to keep the disabled Upload component consistent with the disabled Input as much as possible. How about remove ~pointer-events: none~.
> 
> I found that the `disabled` attribute is missing in `<input type="file">` (upload-content.vue). Maybe adding the `:disabled="disabled"` attribute can avoid using `pointer-events: none`.

I looked at the input component and found that it adds the "is focus" class name when getting focus, while the "focus" selector is used for upload. Do you mean to keep upload consistent with input

<img width="1208" alt="image" src="https://github.com/user-attachments/assets/d8dd1692-fdd9-4a9f-a1ee-9b3c1e2e4334">
<img width="1799" alt="image" src="https://github.com/user-attachments/assets/41b62252-762d-453a-980d-900920280e11">

tolking commented:
I mean the disabled styles of ElInput component.
<img width="138" alt="截屏2024-08-06 13 56 43" src="https://github.com/user-attachments/assets/cf60b3a7-60d1-4194-85df-2895a5319449">

tolking commented on packages/components/upload/src/upload-content.vue:
```diff
@@ -1,7 +1,12 @@
+      ns.is('drag', drag),
+      ns.is('disabled', disabled),
+    ]"
+    :tabindex="disabled ? '-1' : '0'"
```
You also need to add the `disabled` to [L26](https://github.com/element-plus/element-plus/pull/17753/files#diff-848fa2c711dee63acb2665d4ec3ba52674c4644d2e2657bb123d2f0d613a053eR26)

tolking commented on packages/theme-chalk/src/upload.scss:
```diff
@@ -29,6 +29,22 @@
   cursor: pointer;
   outline: none;
 
+  &.is-disabled{
+    &:focus{
```
```suggestion
  &.is-disabled{
    cursor: not-allowed;
    &:focus{
```

ma-shuo committed:
Merge branch 'dev' into dev

ma-shuo commented:
> I mean the disabled styles of ElInput component.

The current code has this effect.
<img width="400" alt="image" src="https://github.com/user-attachments/assets/d1722c92-c75a-4e8a-86c6-d953d84170f8">

tolking reviewed (commented):
Maybe we can also change the background color, WDYT

ma-shuo committed:
style(components): disable style for upload component

Add background color and text color for disabled status and add disabled attribute to input.

closed #17746

ma-shuo committed:
Merge branch 'dev' into dev

ma-shuo committed:
Merge remote-tracking branch 'origin/dev' into dev

tolking reviewed (approved):
Thanks for your contribution.
